File: frontend/src/types/revenue.ts
export interface PipelineStageSummary {
  stage: string;
  count: number;
  unweighted_value: number;
  weighted_value: number;
}

export interface PipelineSummaryResponse {
  total_opportunities: number;
  total_unweighted: number;
  total_weighted: number;
  won_value: number;
  stages: PipelineStageSummary[];
}

export interface RevenueTimelinePoint {
  period: string;
  weighted_value: number;
  won_value: number;
  opportunity_count: number;
}

export interface RevenueTimelineResponse {
  granularity: string;
  points: RevenueTimelinePoint[];
}

export interface AgencyRevenueSummary {
  agency: string;
  opportunity_count: number;
  unweighted_value: number;
  weighted_value: number;
  won_value: number;
}

export interface AgencyRevenueResponse {
  agencies: AgencyRevenueSummary[];
  total_agencies: number;
}

File: backend/app/schemas/revenue.py
"""
Revenue forecasting schemas.
"""

from pydantic import BaseModel


class PipelineStageSummary(BaseModel):
    stage: str
    count: int
    unweighted_value: float
    weighted_value: float


class PipelineSummaryResponse(BaseModel):
    total_opportunities: int
    total_unweighted: float
    total_weighted: float
    won_value: float
    stages: list[PipelineStageSummary]


class RevenueTimelinePoint(BaseModel):
    period: str
    weighted_value: float
    won_value: float
    opportunity_count: int


class RevenueTimelineResponse(BaseModel):
    granularity: str
    points: list[RevenueTimelinePoint]


class AgencyRevenueSummary(BaseModel):
    agency: str
    opportunity_count: int
    unweighted_value: float
    weighted_value: float
    won_value: float


class AgencyRevenueResponse(BaseModel):
    agencies: list[AgencyRevenueSummary]
    total_agencies: int
